client/src/main/java/io/confluent/kafka/schemaregistry/avro/AvroSchema.java

```diff
@@ -576,6 +602,43 @@ private Set<String> getInlineTags(Schema.Field field) {
     return tags;
   }
 
+  private Set<String> getInlineTags(JsonNode tagNode) {
+    Set<String> tags = new LinkedHashSet<>();
+    if (tagNode.has(TAGS)) {
+      ArrayNode tagArray = (ArrayNode) tagNode.get(TAGS);
+      tagArray.elements().forEachRemaining(tag -> tags.add(tag.asText()));
+    }
+    return tags;
+  }
+
+  private void modifyFieldLevelTags(JsonNode node,
+                                    Map<String, Set<String>> tagsToAddMap,
+                                    Map<String, Set<String>> tagsToRemoveMap) {
+    Set<String> pathToModify = new HashSet<>(tagsToAddMap.keySet());
+    pathToModify.addAll(tagsToRemoveMap.keySet());
+
+    for (String path : pathToModify) {
+      JsonNode fieldNodePtr = AvroSchemaUtils.findMatchingField(node, path);
+      Set<String> allTags = getInlineTags(fieldNodePtr);
+
+      if (tagsToAddMap.containsKey(path)) {
+        Set<String> tagsToAdd = tagsToAddMap.get(path);
+        allTags.addAll(tagsToAdd);
+      }
+
+      if (tagsToRemoveMap.containsKey(path)) {
```

Comment:
nit:  it's a little more efficient to call `get` before the `if` block and then have the `if` block check if the result of `get` is not null